kolibri_instant_schools_plugin/management/commands/migratev1database.py

```diff
@@ -0,0 +1,188 @@
+                channel_id=old_log.channel_id,
+                start_timestamp=old_log.start_timestamp,
+                end_timestamp=old_log.end_timestamp,
+                time_spent=old_log.time_spent,
+                progress=old_log.progress,
+                kind=old_log.kind,
+                extra_fields=old_log.extra_fields,
+            )
+            SESSIONLOG_ID_MAPPING[old_log.id] = new_log.id
+
+        # MIGRATE: UserSessionLog
+        for old_log in old_db_session.query(UserSessionLog_).all():
+            new_log, _ = UserSessionLog.objects.get_or_create(
+                user_id=USER_ID_MAPPING[old_log.user_id],
+                channels=old_log.channels,
+                start_timestamp=old_log.start_timestamp,
+                last_interaction_timestamp=old_log.last_interaction_timestamp,
+                pages=old_log.pages,
+            )
+
+        # MIGRATE: MasteryLog
+        for old_log in old_db_session.query(MasteryLog_).all():
+            summarylog = ContentSummaryLog.objects.get(id=SUMMARYLOG_ID_MAPPING[old_log.summarylog_id])
+            new_log, _ = MasteryLog.objects.get_or_create(
+                user_id=summarylog.user_id,
+                summarylog_id=summarylog.id,
+                mastery_criterion=old_log.mastery_criterion,
+                start_timestamp=old_log.start_timestamp,
+                end_timestamp=old_log.end_timestamp,
+                completion_timestamp=old_log.completion_timestamp,
+                mastery_level=old_log.mastery_level,
+                complete=old_log.complete,
+            )
+            MASTERYLOG_ID_MAPPING[old_log.id] = new_log.id
+
+        # MIGRATE: AttemptLog
+        for old_log in old_db_session.query(AttemptLog_).all():
+            sessionlog = ContentSessionLog.objects.get(id=SESSIONLOG_ID_MAPPING[old_log.sessionlog_id])
+            new_log, _ = AttemptLog.objects.get_or_create(
+                user_id=sessionlog.user_id,
+                item=old_log.item,
+                start_timestamp=old_log.start_timestamp,
+                end_timestamp=old_log.end_timestamp,
+                completion_timestamp=old_log.completion_timestamp,
+                time_spent=old_log.time_spent,
+                complete=old_log.complete,
+                correct=old_log.correct,
+                hinted=old_log.hinted,
+                answer=old_log.answer,
+                simple_answer=old_log.simple_answer,
+                interaction_history=old_log.interaction_history,
+                masterylog_id=MASTERYLOG_ID_MAPPING[old_log.masterylog_id],
+                sessionlog_id=sessionlog.id,
+            )
+
+        # CREATE: DeviceSettings
+        device_settings, _ = DeviceSettings.objects.get_or_create()
+        device_settings.is_provisioned = True
+        device_settings.save()
+
+    def load_database(self, path):
+
+        engine = create_engine("sqlite:///" + path, convert_unicode=True)
+        
+        metadata = MetaData()
```

Comment:
If you wanted you could do this in fewer lines with:
```
Base = automap_base()
Base.prepare(engine, reflect=True)
```
http://docs.sqlalchemy.org/en/latest/orm/extensions/automap.html

The only reason I do the more long winded approach is so that I can load a pickled copy of the metadata, rather than using the reflection at runtime.

However, this works fine!

Reply:
Ah, cool! Will try that.